src/Shared/FileUtilities.cs

```diff
@@ -359,41 +359,73 @@ internal static string FixFilePath(string path)
         /// </summary>
         internal static string MaybeAdjustFilePath(string value, string baseDirectory = "")
         {
+            var comparisonType = StringComparison.Ordinal;
+
             // Don't bother with arrays or properties or network paths, or those that
             // have no slashes.
-            if (NativeMethodsShared.IsWindows || string.IsNullOrEmpty(value) ||
-                value.StartsWith("$(") || value.StartsWith("@(") || value.StartsWith("\\\\") ||
-                value.IndexOfAny(Slashes) == -1)
+            if (Path.DirectorySeparatorChar == '\\' || string.IsNullOrEmpty(value)
```

Comment:
IsWindows and IsUnixLike are very cheap, but since they can be called millions of times here, the string compare they perform internally starts to add up. The alternative is locking and caching the values but not sure if it's worth it just for this method.

Reply:
Wouldn't necessarily have to lock; we could racily recompute it N times to a static `bool?` and it'd be fine as long as N < millions . . .